code/modules/atmospherics/machinery/datum_pipeline.dm

```diff
@@ -42,6 +49,13 @@
 	reconcile_air()
 	//Only react if the mix has changed, and don't keep updating if it hasn't
 	update = air.react(src)
+	CalculateGasmixColor(air)
+
+/datum/pipeline/proc/set_air(datum/gas_mixture/new_air)
+	if(new_air == air)
+		return
+	air = new_air
+	CalculateGasmixColor(air)
```

Comment:
Why don't you use same method of generateing gas overlay as a gas can?

Reply:
Much much more expensive and the window is small enough that the gas wouldn't display well